test/framework/src/main/java/org/elasticsearch/logsdb/datageneration/fields/ObjectFieldDataGenerator.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.logsdb.datageneration.fields;
+
+import org.elasticsearch.core.CheckedConsumer;
+import org.elasticsearch.logsdb.datageneration.DataGeneratorSpecification;
+import org.elasticsearch.logsdb.datageneration.FieldDataGenerator;
+import org.elasticsearch.logsdb.datageneration.FieldType;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class ObjectFieldDataGenerator implements FieldDataGenerator {
+    private final DataGeneratorSpecification specification;
+    private final int depth;
+
+    private final List<ChildField> childFields;
+
+    public ObjectFieldDataGenerator(DataGeneratorSpecification specification, int depth) {
```

Comment:
I would also provide a constructor with the list of `FieldDataGenerator` so that we can use dependency injection and eventually create non-random mappings just passing `FieldDataGenerator` instances in the constructor.

Reply:
That's a cool use case, i haven't considered that.

Reply:
Currently leaf `FieldDataGenerator` implementations are not really designed to be created by humans. It's easy to create a constructor that you suggest but then it's a bit annoying to create e.g. `KeywordFieldDataGenerator`. I think we can support this by creating a dedicated object generator that accepts a list of field names and field types. Majority of logic in current one is related to random generation anyway.